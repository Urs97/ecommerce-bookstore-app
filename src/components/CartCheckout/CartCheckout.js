import React from 'react';
import './CartCheckout.scss';

const CartCheckout = ({ quantity, subtotal }) => {
    const order_total = Number(subtotal) + Number(9.99);

    return (
        <section className="cart-checkout">
            <section className="cart-checkout-grid">
                <h2 className="title">Summary</h2>
                <h3 className="title">Cart Total</h3>
                <span className="divider" />
                <span className="quantity">Quantity</span>
                <span className="quantity-number">{quantity} items</span>
                <span className="subtotal">Subtotal</span>
                <span className="subtotal-price">${subtotal}</span>
                <span className="shipping">Shipping</span>
                <span className="shipping-price">$9.99</span>
                <span className="total">Order Total</span>
                <span className="total-price">${order_total}</span>
                <span className="divider-2" />
                <button className="button">Proceed To Checkout</button>
            </section>
        </section>
    );
};

export default CartCheckout;