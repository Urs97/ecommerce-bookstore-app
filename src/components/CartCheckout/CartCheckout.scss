@import '../../styles/shared.scss';

.cart-checkout {
    padding: 0 $horizontal-content-padding;
    padding-bottom: $vertical-content-padding;
    background-color: #F4F7F9;
    display: flex;
    flex-direction: column;
    align-items: flex-end;

    .cart-checkout-grid {
        padding: 2em 2em;
        background-color: #FFFFFF;
        border-radius: $border-radius;
        display: grid;
        grid-template-areas: 
            "title title"
            "subtitle subtitle"
            "divider divider"
            "quantity quantity-number"
            "subtotal subtotal-price"
            "shipping shipping-price"
            "total total-price"
            "divider-2 divider-2"
            "checkout checkout";
        row-gap: 2em;

        h2 {
            grid-area: title;
            font-size: 2em;
        }

        h3 {
            grid-area: subtitle;
            font-size: 1.75em;
        }

        .divider {
            grid-area: divider;
            border-bottom: 3px solid #D3D3D3;
        }

        .quantity {
            grid-area: quantity;
        }

        .quantity-number {
            grid-area: quantity-number;
            justify-self: flex-end;
        }

        .subtotal {
            grid-area: subtotal;
        }

        .subtotal-price {
            grid-area: subtotal-price;
            justify-self: flex-end;
        }

        .shipping {
            grid-area: shipping;
        }

        .shipping-price {
            grid-area: shipping-price;
            justify-self: flex-end;
        }

        .total {
            grid-area: total;
            font-weight: 600;
        }

        .total-price {
            font-weight: 600;
            grid-area: total-price;
            justify-self: flex-end;
        }

        .divider-2 {
            grid-area: divider-2;
            border-bottom: 3px solid #D3D3D3;
        }

        & button {
            grid-area: checkout;
            background-color: red;
        }

        & button:hover {
            background-color: $yellow;
        }
    }
}

@media (max-width: 1100px) {
    .cart-checkout {
        padding: 0 $horizontal-content-padding-1100px;
        padding-bottom: $vertical-content-padding;
    }
}

@media (max-width: 950px) {
    .cart-checkout {
        padding: 0 $horizontal-content-padding-950px;
        padding-bottom: $vertical-content-padding;
    }
}