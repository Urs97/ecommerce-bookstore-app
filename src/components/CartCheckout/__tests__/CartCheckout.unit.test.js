import React from 'react';
import {shallow} from 'enzyme';
import CartCheckout from '../CartCheckout';

describe('CartCheckout', () => {
    test('renders empty CartCheckout', () => {
        const wrapper = shallow(
            <CartCheckout />
        );
        expect(wrapper).toMatchSnapshot();
    });
});