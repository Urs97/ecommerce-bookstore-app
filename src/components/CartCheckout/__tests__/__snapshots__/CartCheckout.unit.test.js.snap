// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartCheckout renders empty CartCheckout 1`] = `
<section
  className="cart-checkout"
>
  <section
    className="cart-checkout-grid"
  >
    <h2
      className="title"
    >
      Summary
    </h2>
    <h3
      className="title"
    >
      Cart Total
    </h3>
    <span
      className="divider"
    />
    <span
      className="quantity"
    >
      Quantity
    </span>
    <span
      className="quantity-number"
    >
       items
    </span>
    <span
      className="subtotal"
    >
      Subtotal
    </span>
    <span
      className="subtotal-price"
    >
      $
    </span>
    <span
      className="shipping"
    >
      Shipping
    </span>
    <span
      className="shipping-price"
    >
      $9.99
    </span>
    <span
      className="total"
    >
      Order Total
    </span>
    <span
      className="total-price"
    >
      $
      NaN
    </span>
    <span
      className="divider-2"
    />
    <button
      className="button"
    >
      Proceed To Checkout
    </button>
  </section>
</section>
`;
